### PR TITLE
feat: deploy with docker

### PR DESCRIPTION
**Pull request by juzhiyuan**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

___
### New feature or improvement
- Describe the details and related test reports.

Deploy the dashboard with Docker (demo site support)

- [ ] Update CD
- [ ] Update docs https://github.com/apache/apisix-dashboard/issues/707

### CODE REVIEW

File: .github/workflows/deploy-with-docker.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          login-server: apisixacr.azurecr.cn
+          username: ${{ secrets.REGISTRY_USERNAME }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+
+      - name: Build then Deploy
+        run: |
+          docker build -t apisixacr.azurecr.cn/dashboard:${{ github.sha }} .
+          docker push apisixacr.azurecr.cn/dashboard:${{ github.sha }}
+          docker run -d -p 80:8080 -v ${pwd}/docs/examples/docker-conf-example.json:/usr/local/apisix-dashboard/conf/conf.json apisixacr.azurecr.cn/dashboard:${{ github.sha }}
```
Comment (juzhiyuan) [inline .github/workflows/deploy-with-docker.yml:40]: @nic-chen @gxthrj 

Please take a look at this line, I'm not sure how to deploy in Azure, please help me to complete the rest work, thanks!

NOTE: When running docker with this image, we should mount a conf file just like this line does.

Comment (juzhiyuan) [reply .github/workflows/deploy-with-docker.yml:40]: Error:

https://github.com/apache/apisix-dashboard/pull/701/checks?check_run_id=1357706148#step:5:976

Comment (nic-chen) [reply .github/workflows/deploy-with-docker.yml:40]: sure, let's fix this.

Comment (juzhiyuan) [reply .github/workflows/deploy-with-docker.yml]: And please update the related docs, thanks.

Comment (nic-chen) [reply .github/workflows/deploy-with-docker.yml]: will update according the test

**Review by membphis — COMMENTED** [review]
LGTM

File: docs/deploy-with-docker.zh-CN.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ $ docker build -t apisix-dashboard:{$tag} .
 
 - `conf.listen.host` 为容器内监听 IP，必须为 `0.0.0.0`，这样宿主才能访问容器内网络。
 - `conf.listen.port` 为容器内监听端口，默认为 `8080`。如需修改，请同步修改 [Dockerfile](../Dockerfile)。
-- `conf.etcd.endpoints` 为 ETCD 主机列表，多个节点以**英文逗号**连接，请确保容器可以访问到这些主机，例如：示例配置中 `conf.etcd.endpoints` 为 `host.docker.internal` 旨在允许容器访问宿主主机上的网络。
+- `conf.etcd.endpoints` 为 ETCD 主机列表，支持多多个节点，请确保容器可以访问到这些主机，例如：示例配置中 `conf.etcd.endpoints` 为 `host.docker.internal` 旨在允许容器访问宿主主机上的网络。
```
[the remark this answers is not in the document]
Comment (nic-chen) [reply docs/deploy-with-docker.zh-CN.md]: fixed.

**Comment by nic-chen** [comment]
@gxthrj  please help deploy in Cloud. I don't know how to trigger the hook.

File: .github/workflows/deploy-with-docker.yml (hunk 1 above)
Comment (gxthrj) [reply .github/workflows/deploy-with-docker.yml]: now use `docker-conf-example.yaml` instead of `docker-conf-example.json`

**Comment by moonming** [comment]
Where is the dockerfile for user? Looks like all for test and deploy

File: .github/workflows/deploy-with-docker.yml (hunk 1 above)
Comment (gxthrj) [reply .github/workflows/deploy-with-docker.yml]: it works now

**Comment by juzhiyuan** [comment]
any update?

**Comment by nic-chen** [comment]
Dockerfile in `./api` had remove into `api/test/docker`:
https://github.com/apache/apisix-dashboard/pull/753

I think there is nothing confusing about Dockerfile now.

**Comment by juzhiyuan** [comment]
OK then let's merge it now.

**Pull request merged**